### ISSUE TITLE
Why?

### ISSUE THREAD

**Issue by NodeGuy** [AnatoliyGatt/deep-freeze-node#4]
Why did you write this instead of using https://github.com/jsdf/deep-freeze?

**Comment by polomsky** [comment]
@NodeGuy currently better implementation is https://github.com/snovakovic/js-flock (better than this and jsdf)

**Comment by NodeGuy** [comment]
Thank you, @polomsky.

**Comment by anatoliygatt** [comment]
@NodeGuy @polomsky There are bunch of libraries that do exactly the same thing, freezing objects. You can use whatever you think is good/descent. There are no particular reasons why this library exists, but apparently it got its users, so it will remain live on NPM.

**Comment by indiesquidge** [comment]
I would note a particular reason to _not_ use `deep-freeze` or `deep-freeze-strict` and stick with this library instead would be to avoid a very niche bug that has to do with the Google Maps API adding implicit object keys to frozen objects.

## Tl;dr

The way around this, according the Google employee commenting on the [issue](https://issuetracker.google.com/issues/72671048), is to only freeze enumerable properties on objects. This library does that correctly, `deep-freeze` and `deep-freeze-strict` do not.


## Details

Reproducible sandbox that shows the issue in isolation: https://codepen.io/indiesquidge/pen/XZWLKv

(_NOTE: if you replace the `deepFreeze` function in the CodePen with the implementation of `deepFreeze` from this library, you will find that the bug no longer causes an infinite recursion stack overflow error._)

Google Maps Issue Tracker: https://issuetracker.google.com/issues/72671048

**Comment by anatoliygatt** [comment]
@indiesquidge Interesting case! Thanks!

**Comment by Eli-Black-Work** [comment]
Is there any chance of mentioning this on the project's page? I think it might be helpful to people when they're shopping around for a deep freeze library :)
